src/MooLite/plugins/WhisperManager/WhisperManagerPlugin.ts

```diff
@@ -0,0 +1,125 @@
+
+    _messages: ChatMessage[] = [];
+    _conversations: Record<string, Conversation> = {};
+    _activeConversationName: string = "";
+
+    config: PluginConfig[] = [
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide System Messages",
+            description: "Hide system messages from the conversation list",
+            key: "hide-system-messages",
+            value: false,
+        },
+        {
+            type: PluginConfigType.CheckBox,
+            name: "Hide Mod Messages",
+            description: "Hide mod messages from the conversation list",
+            key: "hide-mod-messages",
+            value: false,
+        },
+    ];
+
+    tab: MooLiteTab = {
+        icon: "💬",
+        pluginName: this.name,
+        componentName: "WhisperManagerPluginDisplay",
+        component: markRaw(WhisperManagerPluginDisplay),
+        width: "w-80",
+    };
+
+    public get hideSystemMessages(): boolean {
+        return this.getConfig("hide-system-messages").value;
+    }
+
+    public get hideModMessages(): boolean {
+        return this.getConfig("hide-mod-messages").value;
+    }
+
+    public get activeConversation(): Conversation {
+        return this._conversations[this._activeConversationName];
+    }
+
+    public get activeConversationName(): string {
+        return this._activeConversationName;
+    }
+
+    public get messages(): ChatMessage[] {
+        return this._messages;
+    }
+
+    public get conversations(): Record<string, Conversation> {
+        let conversations: Record<string, Conversation> = this.hideSystemMessages
+            ? (({ System, ...o }) => o)(this._conversations)
+            : this._conversations;
+        conversations = this.hideModMessages ? (({ Mods, ...o }) => o)(conversations) : conversations;
+
+        return conversations;
+    }
+
+    public get game(): Game {
+        return this._game;
+    }
+
+    public populateConversations(): void {
+        if (Object.keys(this._conversations).length > 0) return;
+        for (const message of this._game.chat.whisperHistory) {
+            this._addToConversations(message);
+        }
+    }
+
+    public setActiveConversation(name: string): void {
+        if (!this._conversations[name]) return;
+        this._activeConversationName = name;
+        if (this._conversations[name].unread) {
+            this._conversations[name].toggleUnread();
+        }
+    }
+
+    onChatMessage(message: ChatMessage): void {
+        const { channel, isModMessage, isSystemMessage } = message;
+        const isNonStandardMessage = channel === ChatChannelTypeHrid.Whisper || isModMessage || isSystemMessage;
+
+        if (isNonStandardMessage) {
+            this._addToConversations(message, true);
+            return;
+        }
+
+        this._messages.push(message);
+    }
+
+    _addToConversations(message: ChatMessage, markUnread?: boolean): void {
+        const conversationMessage = message as ConversationMessage;
```

Comment:
You can do something like this to avoid casting and possible future bugs!

```ts
        const conversationMessage: ConversationMessage = {
            ...message,
            isInbound: message.receiverName === this._game.character.name,
        };
```

Reply:
Done!